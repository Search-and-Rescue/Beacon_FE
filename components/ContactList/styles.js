const styles = {
  contactsListContainer: {
    alignItems: "center",
    height: "96%",
    marginTop: 20
  },
  contactsList: {
    backgroundColor: "fff",
    borderColor: "#001028",
    borderRadius: 3,
    borderWidth: 4,
    flex: 1,
    height: 300,
    marginHorizontal: "5%",
    width: "90%"
  },
  backgroundImage: {
    flex: 1,
    width: "100%"
  },
  contactCard: {
    backgroundColor: "#efefef",
    flex: 1,
    flexDirection: "row",
    height: 50,
    marginVertical: 5,
    justifyContent: "space-between"
  },
  contactRemoveBtn: {
    color: "#001028",
    height: 30,
    marginHorizontal: 12,
    marginVertical: 10,
    padding: 5
  },
  contactsName: {
    flex: 1,
    fontFamily: "Futura",
    fontSize: 21,
    paddingVertical: 10
  },
  addItemBtn: {
    borderColor: "black",
    borderWidth: 1,
    padding: 10,
    width: "100%"
  },
  addBtnText: {
    fontSize: 21,
    textAlign: "center"
  }
};

export default styles;