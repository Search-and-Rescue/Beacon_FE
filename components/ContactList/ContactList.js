import React, { Component } from "react";
import { View, Text, ScrollView } from "react-native";
import { deleteContact, getEmergencyContacts } from "../../util/apiCalls";
import { connect } from "react-redux";
import styles from "./styles";
import { setEmergencyContacts } from "../../actions";
import { bindActionCreators } from "redux";

export class ContactList extends Component {
  constructor(props) {
    super(props);
  }

  deleteContact = async id => {
    await deleteContact(id);
    const userInfoContacts = await getEmergencyContacts();
    console.log('in delete contact', userInfoContacts)
  };

  render() {
    const contactCards = this.props.contacts.map(contact => {
      return (
        <View key={contact.id} style={styles.contactCard}>
          <Text
            style={styles.contactRemoveBtn}
            onPress={() => this.deleteContact(contact.id)}
          >REMOVE</Text>
          <Text style={styles.contactsName}>{contact.name}
          </Text>
        </View>
      );
    });

    return (
      <View>
        <View style={styles.contactsListContainer}>
          <ScrollView style={styles.contactsList}>
          {contactCards}
          </ScrollView>
        </View>
      </View>
    );
  }
}

export const mapStateToProps = ({ contacts }) => ({
  contacts
});

export const mapDispatchToProps = dispatch =>
  bindActionCreators({ setEmergencyContacts }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(ContactList);
