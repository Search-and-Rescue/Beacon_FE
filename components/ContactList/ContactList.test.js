import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { ContactList } from './ContactList';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

jest.mock('react-native-gesture-handler', () => 'TouchableOpacity');

describe('ContactList', () => {
  let wrapper;

  beforeEach(() => {
    const mockContacts = [{
      name: "Beans",
      phone: "303-1234",
      email: "beans@gmail.com"
    }];

    wrapper = shallow(<ContactList
      contacts={mockContacts}
       />);
  });

  it('should match the snapshot with all of the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });
});