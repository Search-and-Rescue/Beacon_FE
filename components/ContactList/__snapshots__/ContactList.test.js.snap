// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList should match the snapshot with all of the data passed through 1`] = `
<View>
  <View>
    <View />
    <Text>
      Contact List Page
    </Text>
    <View>
      <Button
        onPress={[Function]}
        title="Go to a Contact"
      />
    </View>
  </View>
</View>
`;
